flixel/input/actions/FlxActionInputAnalog.hx

```diff
@@ -0,0 +1,373 @@
+	 * @param	PixelsPerUnit	How many pixels of movement = 1.0 in analog motion (lower: more sensitive, higher: less sensitive)
+	 * @param	DeadZone	Minimum analog value before motion will be reported
+	 * @param	InvertY	Invert the Y axis
+	 * @param	InvertX	Invert the X axis
+	 */
+	public function new(Trigger:FlxAnalogState, Axis:FlxAnalogAxis = EITHER, PixelsPerUnit:Int = 10, DeadZone:Float = 0.1, InvertY:Bool = false, InvertX:Bool = false)
+	{
+		pixelsPerUnit = PixelsPerUnit;
+		if (pixelsPerUnit < 1)
+			pixelsPerUnit = 1;
+		deadZone = DeadZone;
+		invertX = InvertX;
+		invertY = InvertY;
+		super(FlxInputDevice.MOUSE, -1, cast Trigger, Axis);
+	}
+	
+	override public function update():Void
+	{
+		#if !FLX_NO_MOUSE
+			updateXYPosition(FlxG.mouse.x, FlxG.mouse.y);
+		#end
+	}
+	
+	private function updateXYPosition(X:Float, Y:Float):Void
+	{
+		var xDiff = X - lastX;
+		var yDiff = Y - lastY;
+		
+		lastX = X;
+		lastY = Y;
+		
+		if (invertX) xDiff *= -1;
+		if (invertY) yDiff *= -1;
+		
+		xDiff /= (pixelsPerUnit);
+		yDiff /= (pixelsPerUnit);
+		
+		if (Math.abs(xDiff) < deadZone) xDiff = 0;
+		if (Math.abs(yDiff) < deadZone) yDiff = 0;
+		
+		updateVals(xDiff, yDiff);
+	}
+}
+
+class FlxActionInputAnalogMousePosition extends FlxActionInputAnalog
+{
+	/**
+	 * Mouse input -- X/Y is the mouse's absolute screen position
+	 * @param	Trigger What state triggers this action (MOVED, JUST_MOVED, STOPPED, JUST_STOPPED)
+	 * @param	Axis which axes to monitor for triggering: X, Y, EITHER, or BOTH
+	 */
+	public function new(Trigger:FlxAnalogState, Axis:FlxAnalogAxis = EITHER)
+	{
+		super(FlxInputDevice.MOUSE, -1, cast Trigger, Axis);
+	}
+	
+	override public function update():Void 
+	{
+		#if !FLX_NO_MOUSE
+		updateVals(FlxG.mouse.x, FlxG.mouse.y);
+		#end
+	}
+	
+	override private function updateVals(X:Float, Y:Float):Void
+	{
+		if (X != x)
+		{
+			xMoved.press();
+		}
+		else
+		{
+			xMoved.release();
+		}
+		
+		if (Y != y)
+		{
+			yMoved.press();
+		}
+		else
+		{
+			yMoved.release();
+		}
+		
+		x = X;
+		y = Y;
+	}
+}
+
+class FlxActionInputAnalogGamepad extends FlxActionInputAnalog
+{
+	/**
+	 * Gamepad action input for analog (trigger, joystick, touchpad, etc) events
+	 * @param	InputID "universal" gamepad input ID (LEFT_TRIGGER, RIGHT_ANALOG_STICK, TILT_PITCH, etc)
+	 * @param	Trigger What state triggers this action (MOVED, JUST_MOVED, STOPPED, JUST_STOPPED)
+	 * @param	Axis which axes to monitor for triggering: X, Y, EITHER, or BOTH
+	 * @param	GamepadID specific gamepad ID, or FlxInputDeviceID.FIRST_ACTIVE / ALL
+	 */
+	public function new(InputID:FlxGamepadInputID, Trigger:FlxAnalogState, Axis:FlxAnalogAxis = EITHER, GamepadID:Int = FlxInputDeviceID.FIRST_ACTIVE)
+	{
+		super(FlxInputDevice.GAMEPAD, InputID, cast Trigger, Axis, GamepadID);
+	}
+	
+	override public function update():Void 
+	{
+		if (deviceID != FlxInputDeviceID.ALL)	//analog data is only meaningful on an individual device
```

Comment:
Okay, I replaced it with:

```haxe
if (deviceID == FlxInputDeviceID.ALL)
{
	return; //analog data is only meaningful on an individual device
}
```

And removed the later block. Is that what you meant?